tools/integration-tests/tester/tests/testutil.go

```diff
@@ -95,6 +100,271 @@ func CheckForMessageIds(t *testing.T, peers []*framework.Peer, ids map[string]Da
+
+	// abort if no unspent outputs
+	if len(resp.UnspentOutputs[0].OutputIDs) == 0 {
+		return true, ""
+	}
+
+	// calculate available token in the unspent output
+	var availableValue int64 = 0
+	for _, b := range resp.UnspentOutputs[0].OutputIDs[0].Balances {
+		availableValue += b.Value
+		balanceList = append(balanceList, balance.New(getColorFromString(b.Color), (-1)*b.Value))
+	}
+
+	// abort if no enough tokens
+	if availableValue < sentValue {
+		return true, ""
+	}
+
+	out, err := transaction.OutputIDFromBase58(resp.UnspentOutputs[0].OutputIDs[0].ID)
+	require.NoErrorf(t, err, "Invalid unspent outputs ID on %s", from.String())
+	inputs := transaction.NewInputs([]transaction.OutputID{out}...)
+
+	// prepare outputs
+	outmap := map[address.Address][]*balance.Balance{}
+
+	// set balances
+	outmap[outputAddr] = []*balance.Balance{balance.New(balance.ColorIOTA, sentValue)}
+	if availableValue > sentValue {
+		outmap[outputAddr] = append(outmap[outputAddr], balance.New(balance.ColorNew, availableValue-sentValue))
+	}
+	outputs := transaction.NewOutputs(outmap)
+
+	// sign transaction
+	txn := transaction.New(inputs, outputs).Sign(sigScheme)
+
+	// send transaction
+	txId, err = from.SendTransaction(txn.Bytes())
+	require.NoErrorf(t, err, "Could not send transaction on %s", from.String())
+
+	// update balance list
+	balanceList = append(balanceList, outmap[outputAddr]...)
+	updateBalanceList(addrBalance, balanceList, inputAddr.String(), outputAddr.String(), txId)
+
+	return false, txId
+}
+
+func updateBalanceList(addrBalance map[string]map[balance.Color]int64, balances []*balance.Balance, from, to, txId string) {
+	for _, b := range balances {
+		color := b.Color()
+		value := b.Value()
+		if value < 0 {
+			// update from
+			addrBalance[from][color] += value
+		} else {
+			// update to
+			if color == balance.ColorNew {
+				addrBalance[to][getColorFromString(txId)] = value
+			} else {
+				addrBalance[to][color] += value
+			}
+		}
+	}
+	return
+}
+
+func getColorFromString(colorStr string) (color balance.Color) {
+	if colorStr == "IOTA" {
+		color = balance.ColorIOTA
+	} else {
+		t, _ := transaction.IDFromBase58(colorStr)
+		color, _, _ = balance.ColorFromBytes(t.Bytes())
+	}
+	return
+}
+
+// CheckBalances performs checks to make sure that all peers have the same ledger state.
+func CheckBalances(t *testing.T, peers []*framework.Peer, addrBalance map[string]map[balance.Color]int64) {
+	for _, peer := range peers {
+		for addr, b := range addrBalance {
+			sum := make(map[balance.Color]int64)
+			resp, err := peer.GetUnspentOutputs([]string{addr})
+			require.NoError(t, err)
+			assert.Equal(t, addr, resp.UnspentOutputs[0].Address)
+
+			// calculate the balances of each color coin
+			for _, unspents := range resp.UnspentOutputs[0].OutputIDs {
+				for _, b := range unspents.Balances {
+					color := getColorFromString(b.Color)
+					if _, ok := sum[color]; ok {
```

Comment:
Not 100% sure but this should work without the check for existence I think. Since Go will just use the `0-value` of the corresponding type which will be `0` anyway.